samples/Rewrite/ConditionParser/ConditionTestStringParser.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Rewrite.ConditionParser
+{
+    public class ConditionTestStringParser
+    {
+        private const char Percent = '%';
+        private const char Money = '$';
+        private const char Separator = ' ';
+        private const char Colon = ':';
+        private const char OpenBrace = '{';
+        private const char CloseBrace = '}';
+        public static List<ConditionTestStringSegment> ParseConditionTestString(string testString)
+        {
+            // TODO Create different parsers, regex, condition, etc.
+            if (testString == null)
+            {
+                testString = String.Empty;
+            }
+            var context = new ConditionParserContext(testString);
+            var results = new List<ConditionTestStringSegment>();
+            while (context.Next())
+            {
+                if (context.Current == Percent)
+                {
+                    // check for server request param
+                    if (!context.Next())
+                    {
+                        throw new ArgumentException();
+                    }
+                    if (!ParseConditionParameter(context, results))
+                    {
+                        throw new ArgumentException();
+                    }
+                    // here store the information necessary to do an easy lookup
+                }
+                else if (context.Current == Money)
+                {
+                    context.Next();
+                    context.Mark();
+                    // variable
+                    if (context.Current >= '0' && context.Current <= '9')
+                    {
+                        context.Next();
+                        var ruleVariable = context.Capture();
+                        context.Back();
+                        results.Add(new ConditionTestStringSegment { Type = TestStringType.RuleParameter, Variable = ruleVariable });
+                    } else
+                    {
+                        throw new ArgumentException();
+                    }
+                }
+                else
+                {
+                    if (!ParseLiteral(context, results))
+                    {
+                        throw new ArgumentException();
+                    }
+                }
+            }
+            return results;
+        }
+
+        // pre: will only be called if this is an unescaped '%' 
+        // post: adds an application to the teststring results that will be concatinated on success
+        private static bool ParseConditionParameter(ConditionParserContext context, List<ConditionTestStringSegment> results)
```

Comment:
Rather than returning values through arguments, considering returning an object with two fields: a boolean and a list (that can be null)